content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+- for confidential clients, a `client_assertion` is included, with type `jwt-bearer`, signed using the secret client authentication key
+- the client generates a new DPoP key for the user/session and uses it starting with the PAR request
+- if the auth flow started with an account identifier, the client should pass that starting identifier via the `login_hint` field
+- atproto uses PAR, so the request will be sent as an HTTP POST request to the Authorization Server
+
+The Authorization Server will receive the PAR request and use the `client_id` URL to resolve the client metadata document. The server validates the request and client metadata, then stores information about the session, including binding a DPoP key to the session. The server returns a `request_uri` token to the client, including a DPoP nonce via HTTP header.
+
+The client receives the `request_uri` token and prepares to redirect the user. The client must first persist information about the session to some form of storage. This might be a database (for a web service backend) or web platform storage like IndexedDB (for a browser app). The client then redirects the user via browser to the Authorization Server’s auth endpoint, including the `request_uri` as a URL parameter.
+
+The Authorization Server uses the `request_uri` to look up the earlier Authorization Request parameters, authenticates the user (which might include sign-in or account selection), and prompts the user with the Authorization Interface. The user might refine any granular requested scopes, then approves or rejects the request. The Authorization Server redirects the user back to the `redirect_uri`, which might be a web callback URL (for web clients), or a native app URI.
+
+The client uses URL query parameters (`state` and `iss`) to look up and verify session information and stores the authorization `code`. Using the `code`, the client then makes an initial token request to the Authorization Server’s token endpoint. The client completes the PKCE flow by including the earlier value in the `code_verifier` field. Confidential clients need to include a client assertion JWT in the token request; see the "Confidential Client" section. The Authorization Server validates the request and returns a set of tokens, as well as a `sub` field indicating the account identifier (DID) for this session.
+
+At this point it is critical (mandatory) for all clients to verify that the account identified by the `sub` field is consistent with the Authorization Server `issuer`, either by validating against the originally-supplied account DID, or by resolving the accounts DID to confirm the PDS is consistent with the Authorization Server. See “Identity Authentication” section. The Authorization always returns the scopes approved for the session in the `scopes` field (even if they are the same as the request, as an atproto OAuth profile requirement), which may reflect partial authorization by the user.
```

Comment:
NIT: RFC language - The client must verify the account information identified by the sub field ...

But again I point out that tokens are not for clients, they should be treated as opaque blobs. It's on the relying party to validate, not the client.

Reply:
See above: the `sub` field is not in any token, it is the token response, which is a regular JSON object containing the tokens and additional fields (including `sub`)